### PR TITLE
[SPARK-45934][DOCS] Fix `Spark Standalone` documentation table layout

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR fixes `Spark Standalone` documentation table layout.

### Why are the changes needed?

**BEFORE**
- https://spark.apache.org/docs/3.5.0/spark-standalone.html

**AFTER**
- Spark Standalone
Screenshot 2023-11-15 at 2 40 59 AM

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you review this when you have some time, @yaooqinn ?

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -590,41 +590,41 @@ Some of these include:
 
 See the [configuration page](configuration.html) for information on Spark configurations.  The following configurations are specific to Spark on Kubernetes.
 
-#### Spark Properties
+#### Spark Properties (`spark.kubernetes.` prefix is omitted.)
 
 <table class="table table-striped">
 <thead><tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr></thead>
 <tr>
-  <td><code>spark.kubernetes.context</code></td>
```
Comment (yaooqinn) [inline docs/running-on-kubernetes.md:598]: This might break the doc search?

Comment (dongjoon-hyun) [reply docs/running-on-kubernetes.md:598]: That's true. Let me split `kubernetes` part from this PR, @yaooqinn .

**Comment by dongjoon-hyun** [comment]
Could you review this PR, @bjornjorgensen ?

File: docs/running-on-kubernetes.md [hunk 2]
```diff
@@ -1203,17 +1203,17 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td><code>spark.kubernetes.memoryOverheadFactor</code></td>
   <td><code>0.1</code></td>
   <td>
-    This sets the Memory Overhead Factor that will allocate memory to non-JVM memory, which includes off-heap memory allocations, non-JVM tasks, various systems processes, and <code>tmpfs</code>-based local directories when <code>local.dirs.tmpfs</code> is <code>true</code>. For JVM-based jobs this value will default to 0.10 and 0.40 for non-JVM jobs.
+    This sets the Memory Overhead Factor that will allocate memory to non-JVM memory, which includes off-heap memory allocations, non-JVM tasks, various systems processes, and <code>tmpfs</code>-based local directories when <code>spark.kubernetes.local.dirs.tmpfs</code> is <code>true</code>. For JVM-based jobs this value will default to 0.10 and 0.40 for non-JVM jobs.
     This is done as non-JVM tasks need more non-JVM heap space and such tasks commonly fail with "Memory Overhead Exceeded" errors. This preempts this error with a higher default.
     This will be overridden by the value set by <code>spark.driver.memoryOverheadFactor</code> and <code>spark.executor.memoryOverheadFactor</code> explicitly.
```
Comment (bjornjorgensen) [inline docs/running-on-kubernetes.md:1211]: should spark.executor.memoryOverheadFactor this be spark.kubernetes.executor.memoryOverheadFactor ?

Comment (dongjoon-hyun) [reply docs/running-on-kubernetes.md:1211]: This is `Spark Standalone` documetation, @bjornjorgensen ~ 😄

Comment (dongjoon-hyun) [reply docs/running-on-kubernetes.md:1211]: It seems that you are looking at the first commit. I removed K8s part from this PR completely at the latest commit.

Comment (dongjoon-hyun) [reply docs/running-on-kubernetes.md:1211]: Here.
```
$ git diff HEAD~2 --stat
 docs/spark-standalone.md | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)
```

Comment (bjornjorgensen) [reply docs/running-on-kubernetes.md:1211]: yes, I did read the K8s part.

**Comment by bjornjorgensen** [comment]
Thank you for fixing the dokumentasjon for K8S and Standalone :)

**Comment by dongjoon-hyun** [comment]
> Thank you for fixing the dokumentasjon for K8S and Standalone :)

Thanks, but I'm going to proceed K8s part in a new JIRA because of the previous comment.

**Comment by dongjoon-hyun** [comment]
Could you review this `Spark Standalone` documentation PR when you have some time, @huaxingao ?

**Review by huaxingao — APPROVED** [review]

**Comment by huaxingao** [comment]
LGTM Thanks @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Thank you so much, @huaxingao . Merged to master.

**Comment by dongjoon-hyun** [comment]
Also, thank you, @yaooqinn and @bjornjorgensen , too.

**Comment by dongjoon-hyun** [comment]
I also cherry-picked this to branch-3.5.

**Pull request closed without merging**